From: ThirdReality <dev@3reality.com>
Subject: [PATCH] board: amlogic: add ThirdReality HubV3A 1GB DTS

HubV3A used trhub_v3_defconfig, whose DTS hardcodes 2GB in /memory and
mainline dram_init() trusts it. Add a v3a DTS that includes v3 and sets
/memory to 1GB (2x DDR3, 2 ranks), plus trhub_v3a_defconfig.
Applies on top of 0012 (v3l).
---
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -207,6 +207,7 @@
 	meson-axg-s400.dtb \
 	meson-axg-jethome-jethub-j100.dtb \
 	meson-axg-thirdreality-trhub-v3.dtb \
+	meson-axg-thirdreality-trhub-v3a.dtb \
 	meson-axg-thirdreality-trhub-v3b.dtb \
 	meson-axg-thirdreality-trhub-v3l.dtb \
 	meson-gxbb-kii-pro.dtb \
--- a/arch/arm/dts/meson-axg-thirdreality-trhub-v3a.dts
+++ b/arch/arm/dts/meson-axg-thirdreality-trhub-v3a.dts
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * ThirdReality HubV3A: 2x DDR3 16bit 2 ranks = 1GB, no Thread UART.
+ * Board code (LED/GPIO init) is still the HubV3 one, as before this file.
+ */
+
+#include "meson-axg-thirdreality-trhub-v3.dts"
+
+/ {
+	compatible = "thirdreality,trhub-v3a", "amlogic,a113x", "amlogic,meson-axg";
+	model = "ThirdReality TRHub-V3A";
+
+	memory@0 {
+		reg = <0x0 0x0 0x0 0x40000000>;
+	};
+};
--- a/configs/trhub_v3a_defconfig
+++ b/configs/trhub_v3a_defconfig
@@ -0,0 +1,77 @@
+CONFIG_ARM=y
+CONFIG_SYS_BOARD="trhub-v3"
+CONFIG_SYS_CONFIG_NAME="trhub"
+CONFIG_ARCH_MESON=y
+CONFIG_TEXT_BASE=0x01000000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_HAS_CUSTOM_SYS_INIT_SP_ADDR=y
+CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x20000000
+CONFIG_ENV_SIZE=0x2000
+CONFIG_DM_GPIO=y
+CONFIG_DEFAULT_DEVICE_TREE="meson-axg-thirdreality-trhub-v3a"
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_DM_RESET=y
+CONFIG_MESON_AXG=y
+CONFIG_DEBUG_UART_BASE=0xff803000
+CONFIG_DEBUG_UART_CLOCK=24000000
+CONFIG_IDENT_STRING=" trhubv3a"
+CONFIG_SYS_LOAD_ADDR=0x01000000
+CONFIG_DEBUG_UART=y
+CONFIG_REMAKE_ELF=y
+CONFIG_OF_BOARD_SETUP=y
+# CONFIG_DISPLAY_CPUINFO is not set
+CONFIG_MISC_INIT_R=y
+CONFIG_SYS_MAXARGS=32
+# CONFIG_CMD_BDI is not set
+# CONFIG_CMD_IMI is not set
+CONFIG_CMD_EEPROM=y
+CONFIG_CMD_ADC=y
+CONFIG_CMD_GPIO=y
+CONFIG_RANDOM_UUID=y
+CONFIG_CMD_I2C=y
+# CONFIG_CMD_LOADS is not set
+CONFIG_CMD_MMC=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_USB_MASS_STORAGE=y
+CONFIG_CMD_RTC=y
+CONFIG_CMD_REGULATOR=y
+CONFIG_PARTITION_TYPE_GUID=y
+CONFIG_OF_CONTROL=y
+CONFIG_SYS_RELOC_GD_ENV_ADDR=y
+CONFIG_SARADC_MESON=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_MESON=y
+CONFIG_MMC_MESON_GX=y
+CONFIG_MTD_UBI=y
+CONFIG_PHY_REALTEK=y
+CONFIG_ETH_DESIGNWARE_MESON8B=y
+CONFIG_MESON_GXL_USB_PHY=y
+CONFIG_PINCTRL=y
+CONFIG_PINCTRL_MESON_AXG=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_RTC=y
+CONFIG_RTC_PCF8563=y
+CONFIG_DEBUG_UART_ANNOUNCE=y
+CONFIG_DEBUG_UART_SKIP_INIT=y
+CONFIG_MESON_SERIAL=y
+CONFIG_USB=y
+CONFIG_DM_USB_GADGET=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_XHCI_DWC3=y
+CONFIG_USB_DWC2=y
+CONFIG_USB_DWC3=y
+# CONFIG_USB_DWC3_GADGET is not set
+CONFIG_USB_DWC3_MESON_GXL=y
+CONFIG_USB_GADGET=y
+CONFIG_USB_GADGET_VENDOR_NUM=0x1b8e
+CONFIG_USB_GADGET_PRODUCT_NUM=0xfada
+CONFIG_USB_GADGET_DWC2_OTG=y
+CONFIG_USB_GADGET_DOWNLOAD=y
+
+CONFIG_AML_DRIVER=y
+CONFIG_UNIFY_KEY_MANAGE=y
+CONFIG_SECURE_STORAGE=y
+CONFIG_AML_STORAGE=y
+CONFIG_EFUSE=y
+CONFIG_AML_PARTITION=y
